src/main/java/org/prebid/server/bidder/ogury/OguryBidder.java

```diff
@@ -0,0 +1,230 @@
+    }
+
+    private Imp modifyImp(Imp imp, BidRequest bidRequest, ObjectNode impExtBidderHoist) {
+        final Price price = resolvePrice(imp, bidRequest);
+        return imp.toBuilder()
+                .tagid(imp.getId())
+                .bidfloor(price.getValue())
+                .bidfloorcur(price.getCurrency())
+                .ext(modifyExt(imp.getExt(), impExtBidderHoist))
+                .build();
+    }
+
+    private Price resolvePrice(Imp imp, BidRequest bidRequest) {
+        final Price initialBidFloorPrice = Price.of(imp.getBidfloorcur(), imp.getBidfloor());
+        return BidderUtil.shouldConvertBidFloor(initialBidFloorPrice, BIDDER_CURRENCY)
+                ? convertBidFloor(initialBidFloorPrice, bidRequest)
+                : initialBidFloorPrice;
+    }
+
+    private Price convertBidFloor(Price bidFloorPrice, BidRequest bidRequest) {
+        final BigDecimal convertedPrice = currencyConversionService.convertCurrency(
+                bidFloorPrice.getValue(),
+                bidRequest,
+                bidFloorPrice.getCurrency(),
+                BIDDER_CURRENCY);
+
+        return Price.of(BIDDER_CURRENCY, convertedPrice);
+    }
+
+    private ObjectNode modifyExt(ObjectNode impExt, ObjectNode impExtBidderHoist) {
+        if (impExt == null || impExtBidderHoist == null) {
+            return impExt;
+        }
+
+        final ObjectNode modifiedImpExt = impExt.deepCopy();
+        impExtBidderHoist.fieldNames().forEachRemaining(field ->
+                modifiedImpExt.set(field, impExtBidderHoist.get(field)));
+        modifiedImpExt.remove(EXT_FIELD_BIDDER);
+
+        return modifiedImpExt;
+    }
+
+    private boolean hasOguryParams(ObjectNode impExtBidderHoist) {
+        return impExtBidderHoist != null
+                && impExtBidderHoist.has(PREBID_FIELD_ASSET_KEY)
+                && impExtBidderHoist.has(PREBID_FIELD_ADUNIT_ID);
+    }
+
+    private boolean validateRequestKeys(BidRequest request, List<Imp> impsWithOguryParams) {
+        return !CollectionUtils.isEmpty(impsWithOguryParams) && Optional.ofNullable(request.getSite())
+                .map(Site::getPublisher)
+                .map(Publisher::getId)
+                .isEmpty();
+    }
+
+    private MultiMap resolveHeaders(Device device) {
+        final MultiMap headers = HttpUtil.headers().add(HttpUtil.X_OPENRTB_VERSION_HEADER, "2.5");
+
+        if (device != null) {
+            HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.USER_AGENT_HEADER, device.getUa());
+            HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.ACCEPT_LANGUAGE_HEADER, device.getLanguage());
+            HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.X_FORWARDED_FOR_HEADER, device.getIp());
+            HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.X_FORWARDED_FOR_HEADER, device.getIpv6());
+        } else {
+            HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.ACCEPT_LANGUAGE_HEADER, "en-US");
+        }
```

Comment:
why are you trying to add this header if there is no device? I don't see such logic in GO